compiler/rustc_session/src/filesearch.rs

```diff
@@ -80,6 +81,39 @@ fn current_dll_path() -> Result<PathBuf, String> {
         let os = OsStr::from_bytes(bytes);
         Ok(PathBuf::from(os))
     }
+
+    #[cfg(target_os = "aix")]
+    unsafe {
+        let addr = current_dll_path as u64;
+        let mut buffer = vec![0i8; 4096];
```

Comment:
This may not be properly aligned for `ld_info` (it probably will in practice but we shouldn't rely on it). You should use a `vec![std::mem::zeroed::<libc::ld_info>(); N]` instead (and make sure to properly translate the length when calling `loadquery`.

Reply:
That's a good point.